app/src/main/java/com/example/next_show/data/ShowFilter.java

```diff
@@ -0,0 +1,169 @@
+package com.example.next_show.data;
+
+import android.util.Log;
+
+import com.example.next_show.models.Show;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+public class ShowFilter {
+    // instance vars
+    private HashMap<String, List<String>> filters;
+    private List<Show> allShows; // contains all shows from Trakt, THIS IS NOT FILTERED
+    private List<Show> showsList;
+
+    // genres for shows
+    public static final String ACTION = "action";
+    public static final String COMEDY = "comedy";
+    public static final String DRAMA = "drama";
+    public static final String ALL_SHOWS = "all";
+
+    // filters for network
+    public static final String CABLE = "cable";
+    public static final String STREAMING = "streaming";
+    public static final List<String> STREAMING_SERVICES =
+            Arrays.asList("Netflix", "Hulu", "Disney+", "HBOMax", "Peacock", "Amazon", "Apple TV+");
+
+    // filters for year
+    public static final String THIS_YEAR = "2021";
+    public static final String PAST_FIVE = "2016 - 2020";
+    public static final String PRIOR = "< 2015";
+
+    public static final int CURRENT_YEAR = 2021;
+    public static final int PAST_YEAR = 2020;
+    public static final int PRIOR_YEAR = 2015;
+
+    // filters
+    public static final String GENRE = "genre";
+    public static final String NETWORK = "network";
+    public static final String YEAR = "year"; // year aired!
+
+    public ShowFilter(List<Show> showsList) {
+        // when new Filterer is init, set it up
+        this.showsList = showsList;
+        setUpFilters();
+    }
+
+    public void setUpFilters() {
+        filters = new HashMap<>();
+
+        // initialize dictionary with keys and blank lists
+        filters.put(GENRE, new ArrayList<>());
+        filters.put(NETWORK, new ArrayList<>());
+        filters.put(YEAR, new ArrayList<>());
```

Comment:
Hm, do you think there's another data structure instead of array lists that you might want to use here?